buildSrc/src/main/kotlin/codegen/GenerateKotlinDependencyExtensions.kt

```diff
@@ -74,10 +74,10 @@ fun RepositoryHandler.gradleScriptKotlin(): ArtifactRepository =
  * Builds the dependency notation for the named Kotlin [module] at the given [version].
  *
  * @param module simple name of the Kotlin module, for example "reflect".
- * @param version optional desired version, null implies [embeddedKotlinVersion].
+ * @param version optional desired version, unspecified if null.
  */
 fun DependencyHandler.kotlin(module: String, version: String? = null): Any =
-    "org.jetbrains.kotlin:kotlin-${'$'}module:${'$'}{version ?: embeddedKotlinVersion}"
+    "org.jetbrains.kotlin:kotlin-${'$'}module${'$'}{if(version == null) "" else ":${'$'}version"}"
```

Comment:
What's with the wonky syntax here?

Reply:
Hi Jonathan,

This form of [string template escaping](https://kotlinlang.org/docs/reference/basic-types.html#string-templates) is necessary here because this is code generating code that includes a string template.

Were you referring to something else?

Reply:
Oh, okay, I kind of understand.
Why are you using `${'$'}` instead of simply doing `\$`?

Reply:
Both links ☝️  explain that you must use this escaping style when using raw strings (aka. multi-line strings), which is the case here.

Reply:
Ah, fair enough. That makes much more sense.